shimmingtoolbox/coils/siemens_basis.py

```diff
@@ -0,0 +1,154 @@
+    The function first wraps ``shimmingtoolbox.coils.spherical_harmonics`` to generate 1st and 2nd order spherical
+    harmonic ``basis`` fields at the grid positions given by arrays ``X,Y,Z``. *Following Siemens convention*, ``basis`` is
+    then:
+
+        - Reordered along the 4th dimension as *X, Y, Z, Z2, ZX, ZY, X2-Y2, XY*
+
+        - Rescaled to Hz/unit-shim, where "unit-shim" refers to the measure displayed in the Adjustments card of the
+          Syngo console UI, namely:
+
+            - 1 micro-T/m for *X,Y,Z* gradients (= 0.042576 Hz/mm)
+            - 1 micro-T/m^2 for 2nd order terms (= 0.000042576 Hz/mm^2)
+
+    The returned ``basis`` is thereby in the form of ideal "shim reference maps", ready for optimization.
+
+    Args:
+        orders (numpy.ndarray): **not yet implemented** Degrees of the desired terms in the series expansion, specified
+                                as a vector of non-negative integers (``[0:1:n]`` yields harmonics up to n-th order)
+        x (numpy.ndarray): 3-D arrays of grid coordinates, "Right->Left" grid coordinates in the patient coordinate
+                           system (i.e. DICOM reference, units of mm)
+        y (numpy.ndarray): 3-D arrays of grid coordinates (same shape as x). "Anterior->Posterior" grid coordinates in
+                           the patient coordinate system (i.e. DICOM reference, units of mm)
+
+        z (numpy.ndarray): 3-D arrays of grid coordinates (same shape as x). "Inferior->Superior" grid coordinates in
+                           the patient coordinate system (i.e. DICOM reference, units of mm)
+
+    Returns:
+        numpy.ndarray: 4-D array of spherical harmonic basis fields
+
+    NOTES:
+        For now, ``orders`` is, in fact, ignored: fixed as [1:2]—which is suitable for the Prisma (presumably other
+        Siemens systems as well) however, the 3rd-order shims of the Terra should ultimately be accommodated too.
+        (Requires checking the Adjustments/Shim card to see what the corresponding terms and values actually are). So,
+        for now, ``basis`` will always be returned with 8 terms along the 4th dim.
+    """
+
+    # Local functions
+    def reorder_to_siemens(spher_harm):
+        """
+        Reorder 1st - 2nd order basis terms along 4th dim. From
+        1. Y, Z, X, XY, ZY, Z2, ZX, X2 - Y2 (output by shimmingtoolbox.coils.spherical_harmonics.spherical_harmonics), to
+        2. X, Y, Z, Z2, ZX, ZY, X2 - Y2, XY (in line with Siemens shims)
+
+        Args:
+            spher_harm (numpy.ndarray): 4d basis set of spherical harmonics with order/degree ordered along 4th
+                                        dimension. ``spher_harm.shape[3]` must equal 8.
+
+        Returns:
+            numpy.ndarray: 4d basis set of spherical harmonics ordered following siemens convention
+        """
+
+        assert spher_harm.shape[3] == 8
+        reordered = np.zeros(spher_harm.shape)
+        reordered[:, :, :, 0] = spher_harm[:, :, :, 2]
```

Comment:
Instead of doing index by index, you could make a one liner like this:
```suggestion
        reordered = spher_harm[:, :, :, [2, 0, 1, 5, 6, 4, 7, 3]]
```

Reply:
Nice trick